lib/solid_queue/configuration.rb

```diff
@@ -33,12 +33,15 @@ def instantiate
 
     def initialize(**options)
       @options = options.with_defaults(default_options)
+      @worker_id_counter = 0
     end
 
     def configured_processes
-      if only_work? then workers
-      else
-        dispatchers + workers + schedulers
+      @configured_processes ||= begin
+        if only_work? then workers
+        else
+          dispatchers + workers + schedulers
+        end
```

Comment:
Without memoizing this, we would end up with worker ids of `3, 4, 5` instead of `1, 2, 3` in one of the modified tests. Which would be confusing.